src/conduits/RwaInputConduit3.sol

```diff
@@ -197,13 +197,45 @@ contract RwaInputConduit3 {
     //////////////////////////////////*/
 
     /**
-     * @notice Method to swap USDC contract balance to DAI through PSM and push it into RwaUrn address.
+     * @notice Method to swap GEM contract balance to DAI through PSM and push it into RwaUrn address.
      * @dev `msg.sender` must first receive push acess through mate().
      */
     function push() external isMate {
-        uint256 balance = gem.balanceOf(address(this));
+        _doPush(gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Method to swap specific amount of GEM contract balance to DAI through PSM and push it into RwaUrn address.
+     * @dev `msg.sender` must first receive push acess through mate().
+     * @param amt Gem amount
+     */
+    function push(uint256 amt) external isMate {
+        _doPush(amt);
+    }
```

Comment:
⚠️⚠️⚠️ We should either remove `push(uint)` or make it behave the same as the output conduit: not transfer outstanding Dai balance when the amount is specified. ⚠️⚠️⚠️

Reply:
Just thinking that in this case DAI goes to URN, where we can exit this DAI if needed

Reply:
Once the DAI is in the urn, it can only be used to `wipe()` though.

Reply:
Hmm... What about `quit` method in the Urn?

Reply:
Anyway, I already make it works as in OutputConduit